endaq/plot/plots.py

```diff
@@ -445,7 +445,7 @@ def around_peak(df: pd.DataFrame, num: int = 1000, leading_ratio: float = 0.5):
 
     # These can go below and above the number of valid indices, but that can be ignored since
     # they'll only be used to slice the data in a way that is okay to go over/under
-    window_start = max_i - int(num * leading_ratio)
-    window_end = max_i + int(num * (1-leading_ratio))
+    window_start = max(0, max_i - int(num * leading_ratio))
+    window_end = min(len(df)-1, max_i + int(num * (1-leading_ratio)))
```

Comment:
so if you make the threshold at `len(df)-1`, it'll always remove the last element. so this should instead threshold at `len(df)`:
```suggestion
    window_end = min(len(df), max_i + int(num * (1-leading_ratio)))
```

Reply:
really you don't actually need the change at all; you definitely needed the ReLU on the lower bound, because negative indices behave differently, but the upper bound is fine as-is. I know you have feelings about this though so I'll leave it up to you to choose which you want.
```suggestion
    window_end = max_i + int(num * (1-leading_ratio))
```

Reply:
but if there's any chance that this index could be negative, then you should ReLU this too for the same reason as ReLU'ing the lower bound:
```suggestion
    window_end = max(0, max_i + int(num * (1-leading_ratio)))
```